frontend_tests/node_tests/compose_ui.js

```diff
@@ -470,130 +474,59 @@ run_test("markdown_shortcuts", ({override}) => {
-
-    function test_i_typed(isCtrl, isCmd) {
-        // Test 'i' is typed correctly.
-        $("#compose-textarea").val("i");
-        event.key = "i";
-        event.metaKey = isCmd;
-        event.ctrlKey = isCtrl;
-        compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("i", $("#compose-textarea").val());
-    }
 
     function all_markdown_test(isCtrl, isCmd) {
-        input_text = "Any text.";
-        $("#compose-textarea").val(input_text);
-        compose_value = $("#compose-textarea").val();
-        // Select "text" word in compose box.
-        selected_word = "text";
-        range_start = compose_value.search(selected_word);
-        range_length = selected_word.length;
-
         // Test bold:
         // Mac env = Cmd+b
         // Windows/Linux = Ctrl+b
         event.key = "b";
         event.ctrlKey = isCtrl;
         event.metaKey = isCmd;
         compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("Any **text**.", $("#compose-textarea").val());
-        // Test if no text is selected.
-        range_start = 0;
-        // Change cursor to first position.
-        range_length = 0;
-        compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("****Any **text**.", $("#compose-textarea").val());
+        assert.equal(format_text_type, "bold");
+        format_text_type = undefined;
 
         // Test italic:
         // Mac = Cmd+I
         // Windows/Linux = Ctrl+I
         // We use event.key = "I" to emulate user using Caps Lock key.
-        $("#compose-textarea").val(input_text);
-        range_start = compose_value.search(selected_word);
-        range_length = selected_word.length;
         event.key = "I";
         event.shiftKey = false;
         compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("Any *text*.", $("#compose-textarea").val());
-        // Test if no text is selected.
-        range_length = 0;
-        // Change cursor to first position.
-        range_start = 0;
-        compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("**Any *text*.", $("#compose-textarea").val());
+        assert.equal(format_text_type, "italic");
+        format_text_type = undefined;
 
         // Test link insertion:
         // Mac = Cmd+Shift+L
         // Windows/Linux = Ctrl+Shift+L
-        $("#compose-textarea").val(input_text);
-        range_start = compose_value.search(selected_word);
-        range_length = selected_word.length;
         event.key = "l";
         event.shiftKey = true;
         compose_ui.handle_keydown(event, $("#compose-textarea"));
-        assert.equal("Any [text](url).", $("#compose-textarea").val());
-        // Test if exec command is not enabled in browser.
-        queryCommandEnabled = false;
-        compose_ui.handle_keydown(event, $("#compose-textarea"));
+        assert.equal(format_text_type, "link");
+        format_text_type = undefined;
```

Comment:
Thanks for doing this refactoring on the tests; it deletes a lot of message testing code.